sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SqlQuerySpecLogger.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.models.SqlQuerySpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SqlQuerySpecLogger {
+
+    static final String LINE_SEPARATOR = System.getProperty("line.separator");
+    private static final SqlQuerySpecLogger INSTANCE = new SqlQuerySpecLogger(LoggerFactory.getLogger(SqlQuerySpecLogger.class));
+
+    public static SqlQuerySpecLogger getInstance() {
+        return INSTANCE;
+    }
+
+    private final Logger logger;
+
+    SqlQuerySpecLogger(Logger logger) {
+        this.logger = logger;
+    }
+
+    public void logQuery(SqlQuerySpec querySpec) {
+        if (logger.isTraceEnabled() && !querySpec.getParameters().isEmpty()) {
+            StringBuilder queryLogBuilder = new StringBuilder(1000);
+            queryLogBuilder.append(querySpec.getQueryText());
+            querySpec.getParameters().forEach(p -> queryLogBuilder.append(LINE_SEPARATOR)
+                .append(" > param: ")
+                .append(p.getName())
+                .append(" = ")
+                .append(p.getValue(Object.class))
```

Comment:
also it would be good if we only log the query text not the parameter values.
why would you need parameter values to be logged?

Reply:
Partly because I was patterning it after hibernate logging
> org.hibernate.SQL=debug logs the query
> org.hibernate.type=trace  logs the bound parameters

And party because they can be extremely useful when tracking down a misbehaving query or stored procedure. 

it's worth nothing that spring-data-mongo also supports logging query parameters
> org.springframework.data.mongodb.core.MongoTemplate=DEBUG